src/App.js

```diff
@@ -89,16 +89,18 @@ function App() {
       return { score: bestScore, move: move };
     }
   }
+
   // This useEffect hook runs whenever the state of squares or result changes.
   useEffect(() => {
     const isComputerTurn = squares.filter(square => square !== null).length % 2 === 1;
-    const result = calculateResult(squares);
+    const currentResult = calculateResult(squares);
     
-    
-    // If there's a result, it updates the result state.
-    if (result) {
-      setResult(result);
+    if (currentResult && !result) {
+      setResult(currentResult);
       setModalIsOpen(true);
+      if (currentResult === 'x') setScore({ ...score, wins: score.wins + 1 });
+      else if (currentResult === 'o') setScore({ ...score, losses: score.losses + 1 });
+      else if (currentResult === 'tie') setScore({ ...score, ties: score.ties + 1 });
```

Comment:
I'm not a fan of this format, i feel like it makes it a little harder to read. Normally with an `if/else` statement you would format it like this:
if (condition) {
  code to run
} else if (another condition) {
  different code to run
}